src/main.rs

```diff
@@ -201,6 +232,112 @@ fn main() -> ! {
     }
 }
 
+fn check_and_process_control_can_frames(
```

Comment:
The use of `and` in a method name is usually a negative code smell. I don't see any checking being done, perhaps just `process_control_can_frames` will suffice.